datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -552,6 +577,25 @@ fn is_compare_op(op: Operator) -> bool {
     )
 }
 
+// The pruning logic is based on the comparing the min/max bounds.
+// Must make sure the two type has order.
+// For example, casts from string to numbers is not correct.
+// Because the "13" is less than "3" with UTF8 comparison order.
+fn support_type_for_prune(from_type: &DataType, to_type: &DataType) -> bool {
+    // TODO: support other data type
```

Comment:
I think it would be good to file a follow on ticket listing the other types (you did so in https://github.com/apache/arrow-datafusion/issues/3414#issuecomment-1242659478)

Reply:
https://github.com/apache/arrow-datafusion/issues/3442